tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -429,6 +429,37 @@ final class TestingServer(
     }
   }
 
+  def assertHighlight(
+      filename: String,
+      query: String,
+      expected: String,
```

Comment:
I think we might be able to avoid `expected` by encoding the `<<` ranges in the `query` string.
```scala
assertHover(
"""
object Main {
  val <<abc>> = 1
  println(<<a@@bc>> + <<abc>>)
}
```

I don't feel strongly about it, you can use the approach you prefer

Reply:
It's one string now, should be simpler to write new tests.